hyperspy/signal.py

```diff
@@ -3299,19 +3304,26 @@ def fft(self, shift=False, **kwargs):
 
         if self.axes_manager.signal_dimension == 0:
             raise AttributeError("Signal dimension must be at least one.")
+        if apodization==True:
+            apodization = 'hann'
+
+        if apodization:
+            im_fft = self.deepcopy().apply_apodization(window=apodization)
```

Comment:
```suggestion
            im_fft = self.apply_apodization(window=apodization, inplace=False)
```
No need for a `deepcopy` here if you don't use `inplace=True`.